### PR TITLE
[#2] Added a rule file and type info for boost::any_cast (master)

### PR DESCRIPTION
**Pull request by korydraughn**
Please review the install location for the new rule file that fixes the issues related to undefined PEPs and the REPF.

### CODE REVIEW

File: packaging/rep_mtime.re [hunk 1]
```diff
@@ -0,0 +1,8 @@
+pep_api_coll_create_post(*INSTANCE_NAME, *COMM, *INPUT) {}
```
Comment (trel) [inline packaging/rep_mtime.re]: maybe a comment on the first line(s) that explain what this is, why it's here/needed, and how to enable it?

File: CMakeLists.txt [hunk 2]
```diff
@@ -62,6 +62,10 @@ install(FILES ${CMAKE_SOURCE_DIR}/packaging/run_update_collection_mtime_test.py
         DESTINATION ${IRODS_HOME_DIRECTORY}/scripts
         PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ)
 
+install(FILES ${CMAKE_SOURCE_DIR}/packaging/rep_mtime.re
```
Comment (trel) [inline CMakeLists.txt]: suggestion for the filename...
`update_collection_mtime.re` ?

File: packaging/rep_mtime.re (hunk 1 above)
Comment (korydraughn) [reply packaging/rep_mtime.re]: sure, i think that's reasonable.

File: CMakeLists.txt (hunk 2 above)
Comment (korydraughn) [reply CMakeLists.txt]: That name is infinitely better. Should the name include something to indicate what it's for?  For example, **update_collection_mtime_bumper.re**. Possible replacements for **bumper**?
- cushion
- fender
- shield
- screen
- guard _(this one stands out to me)_
- shock absorber
- buffer

Comment (trel) [reply CMakeLists.txt]: i don't think it needs more adjective....   that can go in a comment on the first few lines of the file itself - describing it's reason for being.

File: CMakeLists.txt [hunk 3]
```diff
@@ -62,6 +62,10 @@ install(FILES ${CMAKE_SOURCE_DIR}/packaging/run_update_collection_mtime_test.py
...
+        DESTINATION /etc/irods
```
Comment (korydraughn) [inline CMakeLists.txt]: Are we satisfied with this being the install destination of the new rulebase?

Comment (trel) [reply CMakeLists.txt]: yes, that is the correct location for a packaged rulebase.

the only other question would be if it should be a template when we place it on disk (removing chance for collision later with a live file), and then we tell people to install it into place...

File: CMakeLists.txt (hunk 2 above)
Comment (korydraughn) [reply CMakeLists.txt]: ok

File: CMakeLists.txt (hunk 3 above)
Comment (korydraughn) [reply CMakeLists.txt]: we include .re's with storage tiering, right?
if so, im guessing this should follow that pattern. how does storage tiering handle this?

Comment (korydraughn) [reply CMakeLists.txt]: This resolves the NREP, but what about the PREP? Should we include a Python version of this rulebase or let the admins handle this?

Comment (korydraughn) [reply CMakeLists.txt]: I'm starting to think that these should be templates. Storage tiering does not provide any templates, but the PREP does.

I've also gone ahead and added a Python version of the rulebase.  All that's left now is to add documentation about these files.

File: packaging/rep_mtime.re (hunk 1 above)
Comment (korydraughn) [reply packaging/rep_mtime.re]: Preparing to add comments to the templates.

Comment (korydraughn) [reply packaging/rep_mtime.re]: Is it ok to have the templates point to the README since the it talks about the purpose of these files?

Comment (korydraughn) [reply packaging/rep_mtime.re]: Added comments to templates and pointed templates at README as well.

File: packaging/update_collection_mtime.py.template [hunk 4]
```diff
@@ -0,0 +1,38 @@
+# This template defines all PEPs handled by the MTime REP that trigger continuation in the
+# Rule Engine Plugin Framework. This file is provided to keep errors from being generated
+# when using the MTime REP.
+#
+# If your iRODS deployment defines these PEPs already, then usage of this rulebase
+# is not needed.
+#
+# To use:
+#     1. Copy this template and remove the ".template" extension.
+#     2. Add the new rulebase file to the Python REP's "re_rulebase_set" list.
```
Comment (trel) [inline packaging/update_collection_mtime.py.template]: there is no `re_rulebase_set` in python - it assumes a `core.py` and that things are imported from there.

**Comment by korydraughn** [comment]
What do you mean by providing sufficient bumpers for users/admins?

On another note, I can't imagine an admin would run the test for this on a production deployment. If they did, it would most likely cause the server to complain a lot because the test removes the rulebase files on cleanup (in particular, it will remove the rulebase file that does NOT have the **.template** extension).

File: packaging/update_collection_mtime.py.template (hunk 4 above)
Comment (korydraughn) [reply packaging/update_collection_mtime.py.template]: I was wondering if the PREP supported that. I saw in the code that there was a TODO for that.
I have to adjust the test and docs to reflect this now.

**Comment by trel** [comment]
I think the comment in the template files is sufficient (regarding instructions/bumpers).

File: packaging/update_collection_mtime.py.template (hunk 4 above)
Comment (korydraughn) [reply packaging/update_collection_mtime.py.template]: Adjusted README and Python template to reflect this.

**Pull request merged**